src/equiv.js

```diff
@@ -131,33 +131,53 @@ QUnit.equiv = ( function() {
 		},
 
 		"set": function( b, a ) {
-			var aArray, bArray;
+			var innerEq,
+				outerEq = true;
 
-			aArray = [];
-			a.forEach( function( v ) {
-				aArray.push( v );
-			} );
-			bArray = [];
-			b.forEach( function( v ) {
-				bArray.push( v );
+			if ( a.size !== b.size ) {
+				return false;
+			}
+
+			a.forEach( function( aVal ) {
+				innerEq = false;
+
+				b.forEach( function( bVal ) {
+					if ( innerEquiv( bVal, aVal ) ) {
+						innerEq = true;
+					}
+				} );
+
+				if ( !innerEq ) {
+					outerEq = false;
+				}
```

Comment:
Checking Sets and Maps means we _have_ Sets and Maps, so this could be as simple as:
```js
a.forEach( function( aVal ) {
	eq = eq && b.has( aVal );
} );
```
or the faster but more complex
```js
var iterator = a.values(),
	item = iterator.next();
for ( ; !item.done; item = iterator.next() ) {
	if ( !b.has( item ) ) {
		return false;
	}
}
return true;
```
with analogous options for Map (maybe even _identical_ if both use `entries`).